Unhelpful error message when setting configuration for non-existing exporter

**Describe the bug**

When I set configuration for an exporter that does not exist, for example because I make a typo in the exporter name, Zeebe fails to start and logs a very unhelpful error message that gives me no indication on what went wrong or how to fix it.

**To Reproduce**

``` shell
$ docker run --rm -e ZEEBE_BROKER_EXPORTERS_OOPS_ARGS_INDEX_NUMBEROFREPLICAS=1 camunda/zeebe:latest
...
java.lang.IllegalStateException: Failed to execute CommandLineRunner
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:772) [spring-boot-3.0.5.jar:3.0.5]
        at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:753) [spring-boot-3.0.5.jar:3.0.5]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:317) [spring-boot-3.0.5.jar:3.0.5]
        at io.camunda.zeebe.broker.StandaloneBroker.main(StandaloneBroker.java:82) [camunda-zeebe-8.2.4.jar:8.2.4]
Caused by: java.lang.NullPointerException
        at java.util.concurrent.ConcurrentHashMap.putVal(Unknown Source) ~[?:?]
        at java.util.concurrent.ConcurrentHashMap.putIfAbsent(Unknown Source) ~[?:?]
        at java.lang.ClassLoader.getClassLoadingLock(Unknown Source) ~[?:?]
        at jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(Unknown Source) ~[?:?]
        at jdk.internal.loader.BuiltinClassLoader.loadClass(Unknown Source) ~[?:?]
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(Unknown Source) ~[?:?]
        at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:?]
        at io.camunda.zeebe.broker.exporter.repo.ExporterRepository.load(ExporterRepository.java:79) ~[zeebe-broker-8.2.4.jar:8.2.4]
        at io.camunda.zeebe.broker.Broker.buildExporterRepository(Broker.java:147) ~[zeebe-broker-8.2.4.jar:8.2.4]
        at io.camunda.zeebe.broker.Broker.<init>(Broker.java:70) ~[zeebe-broker-8.2.4.jar:8.2.4]
        at io.camunda.zeebe.broker.Broker.<init>(Broker.java:49) ~[zeebe-broker-8.2.4.jar:8.2.4]
        at io.camunda.zeebe.broker.StandaloneBroker.run(StandaloneBroker.java:91) ~[camunda-zeebe-8.2.4.jar:8.2.4]
        at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:769) ~[spring-boot-3.0.5.jar:3.0.5]
        ... 3 more
...
```

**Expected behavior**

This should result in an easy to understand error message or be ignored silently.

--- a/zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java
+++ b/zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java
@@ -21,6 +21,7 @@ import io.camunda.zeebe.broker.system.configuration.ClusterCfg;
 import io.camunda.zeebe.broker.system.configuration.DataCfg;
 import io.camunda.zeebe.broker.system.configuration.DiskCfg.FreeSpaceCfg;
 import io.camunda.zeebe.broker.system.configuration.ExperimentalCfg;
+import io.camunda.zeebe.broker.system.configuration.ExporterCfg;
 import io.camunda.zeebe.broker.system.configuration.SecurityCfg;
 import io.camunda.zeebe.broker.system.configuration.backup.AzureBackupStoreConfig;
 import io.camunda.zeebe.broker.system.configuration.backup.BackupStoreCfg;
@@ -38,6 +39,7 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Set;
 import org.slf4j.Logger;
 
@@ -110,9 +112,26 @@ public final class SystemContext {
 
     validateExperimentalConfigs(cluster, brokerCfg.getExperimental());
 
+    validateExporters(brokerCfg.getExporters());
+
     final var security = brokerCfg.getNetwork().getSecurity();
     if (security.isEnabled()) {
       validateNetworkSecurityConfig(security);
+    }
+  }
+
+  private void validateExporters(final Map<String, ExporterCfg> exporters) {
+    final Set<Entry<String, ExporterCfg>> entries = exporters.entrySet();
+    final var badExportersNames =
+        entries.stream()
+            .filter(entry -> entry.getValue().getClassName() == null)
+            .map(Entry::getKey)
+            .toList();
+
+    if (!badExportersNames.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Expected to find a 'className' configured for the exporter. Couldn't find a valid one for the following exporters "
+              + badExportersNames);
     }
   }
 

--- a/zeebe/broker/src/test/java/io/camunda/zeebe/broker/system/SystemContextTest.java
+++ b/zeebe/broker/src/test/java/io/camunda/zeebe/broker/system/SystemContextTest.java
@@ -14,16 +14,21 @@ import static org.mockito.Mockito.mock;
 import io.atomix.cluster.AtomixCluster;
 import io.camunda.zeebe.broker.client.api.BrokerClient;
 import io.camunda.zeebe.broker.system.configuration.BrokerCfg;
+import io.camunda.zeebe.broker.system.configuration.ExporterCfg;
 import io.camunda.zeebe.broker.system.configuration.backup.BackupStoreCfg.BackupStoreType;
 import io.camunda.zeebe.broker.system.configuration.partitioning.FixedPartitionCfg;
 import io.camunda.zeebe.broker.system.configuration.partitioning.FixedPartitionCfg.NodeCfg;
 import io.camunda.zeebe.broker.system.configuration.partitioning.Scheme;
 import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.test.util.junit.RegressionTest;
 import io.netty.handler.ssl.util.SelfSignedCertificate;
 import java.io.File;
 import java.security.cert.CertificateException;
 import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
@@ -325,6 +330,31 @@ final class SystemContextTest {
     assertThatCode(() -> initSystemContext(brokerCfg))
         .isInstanceOf(InvalidConfigurationException.class)
         .hasMessageContaining("Failed configuring backup store S3");
+  }
+
+  @RegressionTest("https://github.com/camunda/camunda/issues/12678")
+  void shouldThrowExceptionWithInvalidExporters() {
+    // given
+    final var brokerCfg = new BrokerCfg();
+    final List<String> exportersNames = Arrays.asList("unknown", "oops", "nope");
+    final Map<String, ExporterCfg> exporters = HashMap.newHashMap(exportersNames.size());
+
+    for (final String exporterName : exportersNames) {
+      final ExporterCfg exporterCfg = new ExporterCfg();
+      exporterCfg.setClassName(null);
+      exporterCfg.setJarPath("unknown".equals(exporterName) ? null : exporterName);
+      final Map<String, Object> args = HashMap.newHashMap(1);
+      args.put("any_arg", 1);
+      exporterCfg.setArgs(args);
+      exporters.put(exporterName, exporterCfg);
+    }
+    brokerCfg.setExporters(exporters);
+
+    // then
+    assertThatCode(() -> initSystemContext(brokerCfg))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageStartingWith(
+            "Expected to find a 'className' configured for the exporter. Couldn't find a valid one for the following exporters ");
   }
 
   private SystemContext initSystemContext(final BrokerCfg brokerCfg) {

